### PR TITLE
`Development`: Add string localization guidelines to documentation

### PR DESCRIPTION
**Pull request by rstief**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
It was decided in the dev meeting to always use the `jhiTranslate` directive over the `artemisTranslate` pipe.

### Description
Add a guideline to the documentation to always use the directive when possible.

### Steps for Testing
Visit [the generated docs](https://artemis-platform--8132.org.readthedocs.build/en/8132/dev/guidelines/client.html) to see that the documentation is correct.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
unchanged

## Summary by CodeRabbit

- **Documentation**
	- Enhanced guidelines for localizing user-visible strings.
	- Clarified button and link usage in navigation.
	- Provided instructions for integrating icons, text, and labels.
	- Emphasized code style preferences and memory leak prevention.
	- Defined best practices for routes, breadcrumbs, and strict TypeScript checks.
	- Updated chart instantiation guidelines with `ngx-charts`.
	- Added recommendations for responsive layout design.
	- Limited WebSocket subscriptions for improved performance.

### CODE REVIEW

**Review by laurenzfb — CHANGES_REQUESTED** [review]

**Review by farisd16 — DISMISSED** [review]
Documentation looks fine to me

**Review by egekurt123 — DISMISSED** [review]
LGTM

**Review by milljoniaer — DISMISSED** [review]
Thanks for adding this, looks good 👍

**Review by laurenzfb — APPROVED** [review]

File: docs/dev/guidelines/client.rst [hunk 1]
```diff
@@ -102,16 +102,45 @@ Use JSDoc style comments for functions, interfaces, enums, and classes.
...
+2. To display the string in HTML files, use the ``jhiTranslate`` directive or the ``artemisTranslate`` pipe.
+3. To ensure consistency, always choose the directive over the pipe whenever possible.
+
+Do:
+
+.. code-block:: html+ng2
+
+    <span jhiTranslate="global.title"></span>
+
+    <!-- ok, because there is other content in the span aswell -->
```
Comment (rstief) [inline docs/dev/guidelines/client.rst:120]:
```suggestion
    <!-- ok, because there is other content in the span as well -->
```

**Review by laurenzfb — APPROVED** [review]

**Review by RY997 — APPROVED** [review]
Clear and easy to understand!

**Review by milljoniaer — APPROVED** [review]
reapprove

**Review by rabeatwork — APPROVED** [review]

**Pull request merged**